sdk/daml-lf/transaction/src/main/scala/com/digitalasset/daml/lf/crypto/Hash.scala

```diff
@@ -222,24 +223,68 @@ object Hash {
     }
   }
 
-  // Transaction Builder class with builder methods to recursively encode nodes
-  private sealed class TransactionBuilderV1(
+  object TransactionMetadataBuilderV1 {
+    final case class Metadata(
+        actAs: ImmArray[Ref.Party],
+        commandId: Ref.CommandId,
+        submissionId: Ref.SubmissionId,
+        transactionUUID: UUID,
+        mediatorGroup: Int,
+        domainId: String,
+        ledgerEffectiveTime: Option[Time.Timestamp],
+        submissionTime: Time.Timestamp,
+        submissionSeed: Hash,
```

Comment:
I'm not aware that we're shipping the submission seed as part of the confirmation request. We only ship the node seeds. So the receiving nodes wouldn't be able to validate the hash any more if you include the submission seed here.

I did not see that the node hashing already includes the node seed. So I'd recommend to hash the node seeds with the nodes and remove this field here from the metadata. WDYT?

Reply:
Ah ok, then yes let's do that. Although what's the submission seed for then if we can still re-interpret the transaction in the same way without knowing this seed?

Reply:
The submission seed is just there to derive the transaction seed and from then on the node seeds: https://github.com/digital-asset/daml/blob/main/sdk/daml-lf/spec/contract-id.rst#submission-seed
In some way, it's a more compact representation than explicitly enumerating all derived seeds.

Reply:
Ok, so I'll "project" every node seed to each respective node when encoding nodes then.
And for the `disclosedContracts` contracts nodes then, we have no seed for those right?

Reply:
> And for the disclosedContracts contracts nodes then, we have no seed for those right?

No. Node seeds only belong to nodes in a tree. They're pretty meaningless outside the context of their transaction.

Fine print: Disclosed contracts should have a drivers-specific metadata field that contains a different seed :see_no_evil: This seed is needed to authenticate the contract metadata against the contract ID. But IMO it's not necessary to sign this other seed for the transaction because it does not affect the meaning of the contract. It is merely needed to construct the transaction. And fortunately, the EPN gets hold of this metadata via the `ProcessedDisclosedContract`s.